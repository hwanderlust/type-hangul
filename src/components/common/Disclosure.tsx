import React from "react";
import styled from "styled-components";

import { fadeIn } from "../../helpers";

const Container = styled.footer`
  position: absolute;
  bottom: 3vh;
  display: flex;
  flex-direction: column;
  animation: ${fadeIn} 500ms ease-in;
`;
const Line = styled.div`
  align-self: center;
  width: 30vw;
  height: 1px;
  background-color: black;
`;
const Text = styled.p`
  font-family: "Roboto", serif;
  font-size: 12px;
  font-style: italic;
  text-align: center;
`;

function Disclosure() {
  return (
    <Container>
      <Line></Line>
      <Text>Disclosure: The use of Kakao characters and images are not intended for commercial use and only for educational purposes. No intentions of copyright infingement whatsoever.</Text>
    </Container>
  );
}

export default Disclosure;