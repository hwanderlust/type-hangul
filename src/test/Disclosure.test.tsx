import 'jest-styled-components';
import React from 'react';
import renderer from 'react-test-renderer';

import Disclosure from "../components/common/Disclosure";

describe("<Disclosure/>", () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<Disclosure />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
