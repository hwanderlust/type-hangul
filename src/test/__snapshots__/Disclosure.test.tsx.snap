// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Disclosure/> renders correctly 1`] = `
.c0 {
  margin-top: 10vh;
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-animation: hUniol 4000ms ease-in;
  animation: hUniol 4000ms ease-in;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c1 {
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  width: 30vw;
  height: 1px;
  background-color: black;
}

.c2 {
  font-family: 'Roboto',serif;
  font-size: calc(8px + (12 - 8) * ((100vw - 300px) / (1440 - 300)));
  font-style: italic;
  padding: 2vw;
}

@media screen only and (max-width:720px) {
  .c2 {
    max-width: 500px;
  }
}

<footer
  className="c0"
>
  <div
    className="c1"
  />
  <p
    className="c2"
  >
    Disclosure: The use of Kakao characters and images are not intended for commercial use and only for educational purposes. No intentions of copyright infingement whatsoever.
  </p>
</footer>
`;
